src/ir/module-splitting.cpp

```diff
@@ -336,6 +340,34 @@ ModuleSplitter::initExportedPrimaryFuncs(const Module& primary) {
   return functionExportNames;
 }
 
+std::string ModuleSplitter::generateMinimizedExportName() {
+  // Minimized export names don't need to be human readable at all, so use as
+  // much of the space of single-byte UTF-8 unicode code points as possible.
+  // Binaryen does not actually handle unicode well, though, so only use code
+  // points that correspond to easily printable glyphs: 0x20-0x7E.
+  static std::array<uint8_t, 95> chars = ([]() {
```

Comment:
I think it would be good to emit JS-friendly identifiers. That would leave us the flexibility to integrate the modules using JS as needed. This would have a cost (avoid special characters and reserved words in JS), but I think it would be small. And then we could reuse the code from `src/passes/MinifyImportsAndExports.cpp` (`MinifiedNames`) which would be simpler, too?

Reply:
I agree that it would be good to reuse existing name minification, but the code from `MinifyImportsAndExports.cpp` would require serious refactoring to be able to be used here, where not all imports/exports should be minimized and the minimized names should be generated on the fly rather than all at the same time. I would be happy to do that refactoring, but I'm not sure the benefit of emitting JS-friendly identifiers is worth it. These minimized exports are only exported at all to make them available across the module split; they should not be used, inspected, or referenced in any way by JS. I am comfortable requiring any low-level use case that requires hacking around that abstraction to deal with arbitrary non-JS-friendly names.

Reply:
I think we can move just `MinifiedNames` out of there, and use it here? (I mean literally just that one class) I agree most of the logic can't be shared, but just the minified names generator code (not what decides which imports and exports to be minified).

Reply:
Ah, I hadn't read carefully enough to see that `MinifiedNames` was easily separable. Refactoring done in #3909. Will rebase this PR on top of that once that lands.